patches/api/0449-add-number-format-api.patch

```diff
@@ -0,0 +1,222 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..c1d703b8fb949d9a7ea3b78ead0f031eeb446428
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/scoreboard/numbers/StyledFormatImpl.java
+@@ -0,0 +1,11 @@
++package io.papermc.paper.scoreboard.numbers;
++
++import net.kyori.adventure.text.format.NamedTextColor;
++import net.kyori.adventure.text.format.Style;
++
++record StyledFormatImpl(Style style) implements StyledFormat {
++    public static final StyledFormat NO_STYLE = new StyledFormatImpl(Style.empty());
++    public static final StyledFormat SIDEBAR_DEFAULT = new StyledFormatImpl(Style.empty().color(NamedTextColor.RED));
++    public static final StyledFormat PLAYER_LIST_DEFAULT = new StyledFormatImpl(Style.empty().color(NamedTextColor.YELLOW));
++
++}
+diff --git a/src/main/java/org/bukkit/scoreboard/Objective.java b/src/main/java/org/bukkit/scoreboard/Objective.java
+index a193ffabb05160b462dee1ba8f687fdbc84405b6..7c7cebc06c7d16025a8a12421a79fbe50674049d 100644
+--- a/src/main/java/org/bukkit/scoreboard/Objective.java
++++ b/src/main/java/org/bukkit/scoreboard/Objective.java
+@@ -175,4 +175,22 @@ public interface Objective {
+      */
+     @NotNull Score getScoreFor(@NotNull org.bukkit.entity.Entity entity) throws IllegalArgumentException, IllegalStateException;
+     // Paper end - improve scoreboard entries
++
++    // Paper start - number format
++    /*
++     * Gets the number format for this objective's entries (scores)
++     *
++     * @return this objective's number format
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    @Nullable io.papermc.paper.scoreboard.numbers.NumberFormat numberFormat();
++
++    /*
++     * Sets the number format for this objective's entries (scores)
++     *
++     * @param format the number format to set, pass null to reset format to default
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    void numberFormat(@Nullable io.papermc.paper.scoreboard.numbers.NumberFormat format);
++    // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/scoreboard/Score.java b/src/main/java/org/bukkit/scoreboard/Score.java
+index 1eaa9a93f8eff5f18a6cce2d74f21eb19db273c8..722f6f920045f6f85b347270bbb4e08232db1b9d 100644
+--- a/src/main/java/org/bukkit/scoreboard/Score.java
++++ b/src/main/java/org/bukkit/scoreboard/Score.java
+@@ -83,4 +83,25 @@ public interface Score {
+      */
+     void resetScore() throws IllegalStateException;
+     // Paper end
++
++    // Paper start - number format
++    /*
++     * Gets the number format for this score or null if the score has not been set yet
++     *
++     * @return this score's number format
++     * @throws IllegalStateException if the associated objective has been
++     *     unregistered
++     */
++    @Nullable io.papermc.paper.scoreboard.numbers.NumberFormat numberFormat();
++
++    /*
++     * Sets the number format for this score
```

Comment:
Add `.`.

Generally the main block of javadoc should be proper english with the param/return/throws tags allowing phrases